File: cases/trees/api-be/services/be-reliability.env-outside-config.go
// be-reliability/env-outside-config (Go lane) — bad: keyed process-environment reads outside the
// env-config module this corpus declares (config/env.ts, via zzop-attributes.json). good: reading an
// injected value. os.Environ() is deliberately NOT a read site — the producer names the two keyed
// idioms only, the same whole-environment boundary Python draws for bare os.environ.
package services

import "os"

func BadDsn() string {
	return os.Getenv("DATABASE_URL")
}

func BadPort() (string, bool) {
	return os.LookupEnv("PORT")
}

func Good(dsn string) string {
	return dsn
}

File: cases/trees/api-be/services/be-reliability.env-outside-config.cs
// be-reliability/env-outside-config (C# lane) — bad: a keyed environment read outside the declared
// env-config module (config/env.ts, via zzop-attributes.json). good: an injected value, plus the
// whole-environment GetEnvironmentVariables() (plural), which the producer deliberately does not
// treat as a keyed read site.
class BeReliabilityEnvOutsideConfig {
  string BadDsn() {
    return Environment.GetEnvironmentVariable("DATABASE_URL");
  }

  string Good(string dsn) {
    return dsn;
  }

  System.Collections.IDictionary GoodBulk() {
    return Environment.GetEnvironmentVariables();
  }
}
